### PR TITLE
Create C++ Apriltag example

### PR DESCRIPTION
**Pull request by mcm001**
I just copied ramsete from wpilib

Start PhotonCameraWrapper

bump wpilib everywhere

### CODE REVIEW

**Comment by gerth2** [comment]
I can't leave a comment but why did the libphotoncam.so change? WPILib version update?

File: photonlib-cpp-examples/apriltagExample/networktables.json [hunk 1]
```diff
@@ -0,0 +1 @@
+[]
```
Comment (gerth2) [inline photonlib-cpp-examples/apriltagExample/networktables.json:1]: Maybe remove this and put in gitignore? Not sure what best practice is but it's what we've been doing on 1736 so far

File: photonlib-cpp-examples/apriltagExample/simgui-ds.json [hunk 2]
```diff
@@ -0,0 +1,102 @@
+{
```
Comment (gerth2) [inline photonlib-cpp-examples/apriltagExample/simgui-ds.json:1]: Also maybe gitignore?

File: photonlib-cpp-examples/apriltagExample/simgui-window.json [hunk 3]
```diff
@@ -0,0 +1,74 @@
+{
```
Comment (gerth2) [inline photonlib-cpp-examples/apriltagExample/simgui-window.json:1]: Also maybe gitignore?

File: photonlib-cpp-examples/apriltagExample/simgui.json [hunk 4]
```diff
@@ -0,0 +1,37 @@
+{
```
Comment (gerth2) [inline photonlib-cpp-examples/apriltagExample/simgui.json:1]: Also maybe gitignore?

File: photonlib-cpp-examples/apriltagExample/src/main/cpp/Robot.cpp [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+#include <networktables/NetworkTable.h>
+
+void Robot::RobotInit() {
+  if constexpr (frc::RobotBase::IsSimulation()) {
+    auto inst = nt::NetworkTableInstance::GetDefault();
+    inst.StopServer();
+    // Change the IP address in the below function to the IP address you use to
+    // connect to the PhotonVision UI.
+    inst.SetServer("localhost");
+    inst.StartClient4("Robot Simulation");
```
Comment (gerth2) [inline photonlib-cpp-examples/apriltagExample/src/main/cpp/Robot.cpp:37]: nitpick - change the java side too to have a better name for the client

Comment (gerth2) [reply photonlib-cpp-examples/apriltagExample/src/main/cpp/Robot.cpp:37]: clarity - I like this answer better than the java answer of `myRobot`.

**Review by gerth2 — APPROVED** [review]
Looks good with two major caveats:
1) I'm bad at c++
2) not run on a robot.

Plus a few nits about things that might be better to be ignored, at least for example purposes.

But, in terms of communicating the key concepts, it works.

Sim support soon hopefully :). But likely not till after the season.

File: photonlib-cpp-examples/apriltagExample/simgui-ds.json (hunk 2 above)
Comment (mcm001) [reply photonlib-cpp-examples/apriltagExample/simgui-ds.json:1]: I vote no, so I can set up WASD bindings for driving around

**Comment by mcm001** [comment]
WPILib version update is coz we forgot to bump gradlerio

**Comment by gerth2** [comment]
Solid

**Review by gerth2 — APPROVED** [review]

**Pull request merged**